lib/devices/tuya-device.js

```diff
@@ -119,9 +120,13 @@ class TuyaDevice extends Device {
     }
     this.tuyapi.find({timeout: this.adapter.config.timeout}).then((() => {
       console.debug(this.id, 'Device found. Connecting...');
-      this.tuyapi.connect();
-    }).bind(this)).catch((() => {
-      this.run();
+      this.tuyapi.connect().catch(((err) => {
+        console.debug(this.id, 'Connect failed', err);
+        setTimeout(this.run.bind(this), this.adapter.config.timeout * 1000);
+      }).bind(this));
```

Comment:
(`.bind(this)` is a no-op on arrow functions)

Reply:
Completely true, thank you!